File: sdks/php/src/Plans/Requests/AddPricePlansRequest.php
<?php

namespace NovaBilling\Plans\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Types\CreatePlanPriceDto;

class AddPricePlansRequest extends JsonSerializableType
{
    /**
     * @var CreatePlanPriceDto $body
     */
    public CreatePlanPriceDto $body;

    /**
     * @param array{
     *   body: CreatePlanPriceDto,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->body = $values['body'];
    }
}

File: sdks/ruby/lib/novabilling/plans/types/add_price_plans_request.rb
# frozen_string_literal: true

module Novabilling
  module Plans
    module Types
      class AddPricePlansRequest < Internal::Types::Model
        field :id, -> { String }, optional: false, nullable: false
        field :body, -> { Novabilling::Types::CreatePlanPriceDto }, optional: false, nullable: false
      end
    end
  end
end
